spec/system/consumer/split_checkout_spec.rb

```diff
@@ -207,10 +207,56 @@
         end
       end
 
-      describe "filling out delivery details" do
+      describe "filling out order details" do
         before do
           fill_out_details
           fill_out_billing_address
+          order.update(user_id: user.id)
+        end
+
+        context "billing address" do
+          describe "checking the default address box" do
+            before do
+              check "order_save_bill_address"
+              choose free_shipping.name
+              proceed_to_payment
+            end
+
+            it "before saving has no default address" do
+              expect(user.bill_address).to be_nil
+              expect(order.bill_address).to be_nil
+            end
+
+            context "as a first time customer" do
+              it "creates a new default bill address" do
+                expect(order.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(order.customer.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(user.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+              end
+            end
+          end
+
+          describe "unchecking the default address box" do
+            before do
+              uncheck "order_save_bill_address"
+              choose free_shipping.name
+            end
+
+            context "as an existing customer" do
+              let(:existing_address) { create(:address) }
+
+              before do
+                user.bill_address = existing_address
+                user.ship_address = existing_address
+                proceed_to_payment
+              end
+              it "updates the bill address of the order and customer" do
+                expect(order.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(order.customer.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(user.bill_address.address1).to eq(existing_address.address1)
```

Comment:
I have a question about this line:

The spec passes this way, `user.bill_address` is still = `existing_address`:
```
(byebug) user.bill_address.address1
"10 Lovely Street"
```


I would expect it still to be the case if `user.update.bill_address` is used instead. But in this case I get:

```
(byebug) user.update.bill_address.address1
*** ArgumentError Exception: wrong number of arguments (given 0, expected 1)

nil
```

Do we need to `reload` in this case?

Reply:
Added 13b7d8063273c28b4849acd867311b7cad552ca8; I think only after:
- saving user address
- proceeding to payment step
We can confidently reload the user info from the DB and assert that it was not changed.